packages/calcite-components/src/components/radio-button/radio-button.tsx

```diff
@@ -271,11 +274,26 @@ export class RadioButton
     });
   }
 
+  private updateTabIndexOfOtherRadioButtonsInGroup(): void {
+    const radioButtons = this.queryButtons();
+    const otherFocusableRadioButtons = radioButtons.filter(
+      (radioButton) => radioButton.guid !== this.guid && !radioButton.disabled
+    );
+    otherFocusableRadioButtons.forEach((radiobutton) => {
+      forceUpdate(radiobutton);
+    });
+  }
+
   private getTabIndex(): number | undefined {
     if (this.disabled) {
       return undefined;
     }
-    return this.checked || this.isDefaultSelectable() ? 0 : -1;
+    if (this.checked || this.isFocusable()) {
+      this.updateTabIndexOfOtherRadioButtonsInGroup();
```

Comment:
I'm wondering if `updateTabIndexOfOtherRadioButtonsInGroup` should be called in `connectedCallback` instead of on every render?  Seems better from a performance standpoint to only run this when we know the actual DOM has been modified.

Reply:
i don't think moving this to `connectedCallback` is required. This method executed only once per radio-group on initial render because only one radio-button will have the `tabindex` set to `0`. Any new addition or deletion of radio-buttons in the group causing change in `tabIndex` will invoke this method which is likely to happen once per deletion or addition .

Reply:
> This method executed only once per radio-group on initial render

I'm seeing this method getting called on initial load and anytime a new value is selected, so its not just happening on initial render:

![image](https://github.com/Esri/calcite-components/assets/821864/27d9c136-650d-43c8-8c84-334fb8b22df5)

Reply:
Seems to work just fine if I simply move the call to `this.updateTabIndexOfOtherRadioButtonsInGroup()` into `connectedCallback`.  I now only see those forceUpdate calls happening when I add or remove radios.

Reply:
This works as expected and enhances the  performance.